Solid start Reference section (#604)
solidjs/solid-docs-next: src/routes/solid-start/reference/server/use-server.mdx

---
title: '"use server"'
---

Perform actions on the server environment only (i.e. console logging, etc.).

```tsx twoslash
const logHello = async (message: string) => {
	"use server";
	console.log(message);
};
```

## Basic usage

> In the two examples below, regardless of whether we are rendering on the server or in the browser, the `logHello` function generates a log on the server console only. To do this we use compilation to transform the `"use server"` function into an RPC call to the server.

Reviewer: ```suggestion
When using `"use server"`, regardless of whether rendering is happening on the server or in the browser, the function will only work in the server.
To do this, compilation is used to transform the `"use server"` function into an RPC call to the server.
```

If `"use server"` is inserted as the first line in a file, the entire file will become server-only.

> ```tsx twoslash {1}

Reviewer: ```suggestion
```tsx {1}
```

"use server";

const logHello = async (message: string) => {
	console.log(message);
};
```

> If `"use server"` is instead inserted as the first line of a function, only that function will become server-only.

Reviewer: ```suggestion
However, if `"use server"` is inserted as the first line of a function, only that function will be server-only:
```

> ```tsx twoslash {2}

Reviewer: ```suggestion
```tsx {2}
```

const logHello = async (message: string) => {
	"use server";
	console.log(message);
};

logHello("Hello");
```

> 

Reviewer: ```suggestion

In both of these examples, the `logHello` function, it would only show in the server console regardless of whether rendering was on the server or in the browser

```

## Serialization

> Server functions allow the serialization of many different data types in the response, using the Seroval serializer. The full list is available [here](https://github.com/lxsmnsyc/seroval/blob/main/docs/compatibility.md#supported-types).

Reviewer: ```suggestion
Server functions allow the serialization of many different data types in the response, using the Seroval serializer.
The full list is available [in Seroval's source code](https://github.com/lxsmnsyc/seroval/blob/main/docs/compatibility.md#supported-types).
```

---
title: "\"use server\""
---
Perform actions on the server environment only (i.e. console logging, etc.).

```tsx twoslash
// ---cut---
const logHello = async (message: string) => {
  "use server";
  console.log(message);
};
```

## Basic usage

In the two examples below, regardless of whether we are rendering on the server or in the browser, the `logHello` function generates a log on the server console only. To do this we use compilation to transform the `"use server"` function into an RPC call to the server.

If `"use server"` is inserted as the first line in a file, the entire file will become server-only. 

```tsx twoslash {1}
"use server";

const logHello = async (message: string) => {
  console.log(message);
};
```

If `"use server"` is instead inserted as the first line of a function, only that function will become server-only. 

```tsx twoslash {2}
const logHello = async (message: string) => {
  "use server";
  console.log(message);
};

logHello("Hello");
```

> 
> 
> 
> 
> 

Reviewer: ```suggestion

```
